From: Alexander Graul <agraul@suse.com>
Date: Fri, 22 Oct 2021 12:24:13 +0200
Subject: Mock ip_addrs() in utils/minions.py unit test
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Previously the test used `salt.utils.network.ip_addrs()' in the same way
that the tested code did. This worked well as long as at least one IP
address was returned by `salt.utils.network.ip_addrs()'.

Since this is a unit test, it should not depend on the environment,
it should just work™, even if there are no real IP addresses assigned to
the system (or container) that runs the test.

Bug-Debian: https://bugs.debian.org/1005053
Origin: https://github.com/saltstack/salt/pull/60324
Co-authored-by: Pablo Suárez Hernández <psuarezhernandez@suse.com>
---
 tests/pytests/unit/utils/test_minions.py | 27 +++++++++++++++------------
 1 file changed, 15 insertions(+), 12 deletions(-)

diff --git a/tests/pytests/unit/utils/test_minions.py b/tests/pytests/unit/utils/test_minions.py
index 7875a95..cd0bb36 100644
--- a/tests/pytests/unit/utils/test_minions.py
+++ b/tests/pytests/unit/utils/test_minions.py
@@ -9,18 +9,21 @@ def test_connected_ids():
     test ckminion connected_ids when
     local_port_tcp returns 127.0.0.1
     """
-    opts = {"publish_port": 4505, "detect_remote_minions": False}
+    opts = {
+        "publish_port": 4505,
+        "detect_remote_minions": False,
+        "minion_data_cache": True,
+    }
     minion = "minion"
-    ip = salt.utils.network.ip_addrs()
-    mdata = {"grains": {"ipv4": ip, "ipv6": []}}
-    ckminions = salt.utils.minions.CkMinions({"minion_data_cache": True})
-    patch_net = patch("salt.utils.network.local_port_tcp", return_value={"127.0.0.1"})
+    ips = {"203.0.113.1", "203.0.113.2", "127.0.0.1"}
+    mdata = {"grains": {"ipv4": ips, "ipv6": []}}
+    patch_net = patch("salt.utils.network.local_port_tcp", return_value=ips)
     patch_list = patch("salt.cache.Cache.list", return_value=[minion])
     patch_fetch = patch("salt.cache.Cache.fetch", return_value=mdata)
-    with patch.dict(ckminions.opts, opts):
-        with patch_net, patch_list, patch_fetch:
-            ret = ckminions.connected_ids()
-            assert ret == {minion}
+    ckminions = salt.utils.minions.CkMinions(opts)
+    with patch_net, patch_list, patch_fetch:
+        ret = ckminions.connected_ids()
+        assert ret == {minion}
 
 
 def test_connected_ids_remote_minions():
@@ -37,10 +40,10 @@ def test_connected_ids_remote_minions():
     minion = "minion"
     minion2 = "minion2"
     minion2_ip = "192.168.2.10"
-    ip = salt.utils.network.ip_addrs()
-    mdata = {"grains": {"ipv4": ip, "ipv6": []}}
+    minion_ips = {"203.0.113.1", "203.0.113.2", "127.0.0.1"}
+    mdata = {"grains": {"ipv4": minion_ips, "ipv6": []}}
     mdata2 = {"grains": {"ipv4": [minion2_ip], "ipv6": []}}
-    patch_net = patch("salt.utils.network.local_port_tcp", return_value={"127.0.0.1"})
+    patch_net = patch("salt.utils.network.local_port_tcp", return_value=minion_ips)
     patch_remote_net = patch(
         "salt.utils.network.remote_port_tcp", return_value={minion2_ip}
     )
